jsurfer-core/src/main/java/org/jsfr/json/path/JsonPath.java

```diff
@@ -94,7 +95,7 @@ public void remove() {
 
         public static Builder start(SyntaxMode mode) {
             if (mode == SyntaxMode.STRICT) {
-                throw new UnsupportedStateException("Strict JsonPath mode not yet supported");
+                throw new JsonSurfingException("Strict JsonPath mode not yet supported");
```

Comment:
Maybe `JsonPathCompilerException` is more relevant here since this exception occurs on json path compilation stage.